transcendence-app/src/auth/auth.controller.ts

```diff
@@ -20,15 +24,22 @@ export class AuthController {
   @Get('login')
   @ApiOkResponse({ description: 'Login successfully' })
   @ApiFoundResponse({ description: 'Redirect to 42 OAuth server' })
-  @ApiBadGatewayResponse({ description: 'Bad Gateway' })
+  @ApiServiceUnavailableResponse({ description: 'Service unavailable' })
   @UseGuards(OAuth42Guard)
   oauth42Login() {
     // Guard implementation
   }
 
   @Delete('logout')
+  @UseGuards(AuthenticatedGuard)
+  @ApiForbiddenResponse({ description: 'Forbidden' })
   @ApiOkResponse({ description: 'Logout successfully' })
+  @ApiNotFoundResponse({ description: 'Not Found' })
   logout(@GetRequest() req: Request) {
-    req.logout();
+    req.logout(function (err: Error) {
+      if (err) {
+        throw new NotFoundException(err.message);
+      }
+    });
```

Comment:
This method got an update in [passport 0.6.0](https://medium.com/passportjs/fixing-session-fixation-b2b68619c51d) and now is async and has a callback parameter